phoenix-core/src/test/java/org/apache/phoenix/parse/QueryParserTest.java

```diff
@@ -803,6 +803,26 @@ public void testValidUpsertSelectHint() throws Exception {
             parseQuery(sql);
     }
 
+    @Test
+    public void testValidMultipleUpsert() throws Exception {
+        String sql = (
+                (
+                        "upsert into t VALUES(1,2),(3,4)"));
+        parseQuery(sql);
+    }
+
+    @Test
+    public void testValidMultipleUpsert2() throws Exception {
+        String sql = "upsert into t(a,b) VALUES(1,2),(3,4)";
+        parseQuery(sql);
+    }
+
+    @Test
+    public void testValidMultipleUpsert3() throws Exception {
```

Comment:
A few more failure cases to exercise the parser would be nice.
i.e. no value list at all, no commas between, not closing some parentheses and similar.